src/features/session_replay/shared/recorder.js

```diff
@@ -74,8 +85,42 @@ export class Recorder {
     }
   }
 
+  /**
+   * audit - Checks if the event node payload is missing certain attributes
+   * will forward on to the "store" method if nothing needs async fixing
+   * @param {*} event - An RRWEB event node
+   * @param {*} isCheckout - Flag indicating if the payload was triggered as a checkout
+   */
+  audit (event, isCheckout) {
+    /** only run the audit if inline_stylesheets is configured as on (default behavior) */
+    if (this.shouldInlineStylesheets === false || !this.shouldFix) {
+      this.currentBufferTarget.inlinedAllStylesheets = false
+      return this.store(event, isCheckout)
+    }
+    /** An count of stylesheet objects that were blocked from accessing contents via JS */
+    const incompletes = stylesheetEvaluator.evaluate()
+    /** Only stop ignoring data if already ignoring and a new valid snapshap is taking place (0 incompletes and we get a meta node for the snap) */
+    if (!incompletes && this.#fixing && event.type === RRWEB_EVENT_TYPES.Meta) this.#fixing = false
+    if (incompletes) {
+      handle(SUPPORTABILITY_METRIC_CHANNEL, ['SessionReplay/Payload/Missing-Inline-Css', incompletes], undefined, FEATURE_NAMES.metrics, this.parent.ee)
+      /** wait for the evaluator to download/replace the incompletes' src code and then take a new snap */
+      stylesheetEvaluator.fix().then((failedToFix) => {
+        if (failedToFix) {
+          this.currentBufferTarget.inlinedAllStylesheets = false
+          this.shouldFix = false
+        }
+        this.takeFullSnapshot()
```

Comment:
What is the expectation here if we fail to fix the stylesheets?

Reply:
it will take a new snapshot and then stop "trying to fix" the stylesheets.